utils/node_local_test.py

```diff
@@ -1949,11 +1950,21 @@ class needs_dfuse_with_opt():
     to achieve this.
     """
 
-    # pylint: disable=too-few-public-methods
+    # dict of names that have been decorated either by needs_dfuse_with_opt or needs_dfuse
+    # values are list of possible variants
+    wrapped_names = {}
 
-    def __init__(self, caching=None, wbcache=True, single_threaded=False, dfuse_inval=True,
-                 ro=False):
-        self.caching = caching
+    # dict of tests that have been excluded at runtime (so we know what to skip)
+    # values are the versions of the test to skip
+    excluded_name_dict = {}
+
+    # ensure thread safety; not sure this is actually necessary, but meh...
+    wrapping_lock = threading.Lock()
+
+    # pylint: disable=too-few-public-methods
+    def __init__(self, caching_variants=[False, True], wbcache=True, single_threaded=False,
```

Comment:
> dangerous-default-value, Dangerous default value [] as argument

The safe way is
```suggestion
    def __init__(self, caching_variants=None, wbcache=True, single_threaded=False,
```
And then
```
if caching_variants is None:
    caching_variants = [False, True]
```

Reply:
Fixed.